Detections/AzureActivity/Granting_Permissions_To_Account_detection.yaml

```diff
@@ -18,25 +18,40 @@ relevantTechniques:
   - T1098
   - T1548
 query: |
-
   let starttime = 14d;
   let endtime = 1d;
   // The number of operations below which an IP address is considered an unusual source of role assignment operations
   let alertOperationThreshold = 5;
+  let AzureBuiltInRole = externaldata(Role:string,RoleDescription:string,ID:string) [@"https://raw.githubusercontent.com/Azure/Azure-Sentinel/master/Sample%20Data/Feeds/AzureBuiltInRole.csv"] with (format="csv", ignoreFirstRecord=True);
   let createRoleAssignmentActivity = AzureActivity
   | where OperationNameValue =~ "microsoft.authorization/roleassignments/write";
-  createRoleAssignmentActivity 
+  let RoleAssignedActivity = createRoleAssignmentActivity 
   | where TimeGenerated between (ago(starttime) .. ago(endtime))
   | summarize count() by CallerIpAddress, Caller
   | where count_ >= alertOperationThreshold
   | join kind = rightanti ( 
   createRoleAssignmentActivity
   | where TimeGenerated > ago(endtime)
+  | extend PrincipalId = tostring(parse_json(tostring(parse_json(tostring(Properties_d.requestbody)).Properties)).PrincipalId)
+  | extend PrincipalType = tostring(parse_json(tostring(parse_json(tostring(Properties_d.requestbody)).Properties)).PrincipalType)
+  | extend Scope = tostring(parse_json(tostring(parse_json(tostring(Properties_d.requestbody)).Properties)).Scope)
+  | extend RoleAddedDetails = tostring(parse_json(tostring(parse_json(tostring(parse_json(Properties).requestbody)).Properties)).RoleDefinitionId) 
   | summarize StartTimeUtc = min(TimeGenerated), EndTimeUtc = max(TimeGenerated), ActivityTimeStamp = make_set(TimeGenerated), ActivityStatusValue = make_set(ActivityStatusValue), 
   OperationIds = make_set(OperationId), CorrelationId = make_set(CorrelationId), ActivityCountByCallerIPAddress = count()  
-  by ResourceId, CallerIpAddress, Caller, OperationNameValue, Resource, ResourceGroup
+  by ResourceId, CallerIpAddress, Caller, OperationNameValue, Resource, ResourceGroup, PrincipalId, PrincipalType, Scope, RoleAddedDetails
   ) on CallerIpAddress, Caller
-  | extend timestamp = StartTimeUtc, AccountCustomEntity = Caller, IPCustomEntity = CallerIpAddress
+  | extend timestamp = StartTimeUtc, AccountCustomEntity = Caller, IPCustomEntity = CallerIpAddress;
+  let RoleAssignedActivitywithRoleDetails = RoleAssignedActivity
+  | extend RoleAssignedID = tostring(split(RoleAddedDetails, "/")[-1])
+  | join kind = inner (AzureBuiltInRole 
+  ) on $left.RoleAssignedID == $right.ID;
+  RoleAssignedActivitywithRoleDetails
+  | summarize arg_max(StartTimeUtc, *) by PrincipalId, RoleAssignedID
+  | join kind = leftouter( IdentityInfo
```

Comment:
need to update Connector for `IdentityInfo` table

Reply:
Done. Thanks Ashwin